kbchat/kvstore_test.go

```diff
@@ -0,0 +1,102 @@
+package kbchat
+
+import (
+	"testing"
+
+	"github.com/keybase/go-keybase-chat-bot/kbchat/types/keybase1"
+	"github.com/stretchr/testify/require"
+)
+
+func contains(a []string, x string) bool {
+	for _, n := range a {
+		if x == n {
+			return true
+		}
+	}
+	return false
+}
+
+func containsKey(a []keybase1.KVListEntryKey, x string) bool {
+	for _, n := range a {
+		if x == n.EntryKey {
+			return true
+		}
+	}
+	return false
+}
+
+// test doesn't make assumption about prior state
+func TestKVStore(t *testing.T) {
+	var err error
+
+	alice, dir := testBotSetup(t, "alice")
+	channel := getTeamChatChannel(t, "acme")
+	team := channel.Name
+	defer testBotTeardown(t, alice, dir)
+
+	namespace := "_test_namespace1"
+	key := "_test_key1"
+
+	// put with default revision
+	put, err := alice.PutEntry(team, namespace, key, "value1")
+	require.NoError(t, err)
+	require.True(t, put.Revision > 0)
+	rev := put.Revision
+
+	expectedRevision := rev + 1
```

Comment:
this is a little confusing. and `rev` is never referred to (or incremented) anywhere after here. i'd suggest having one variable called `currentRevision`. and then it's super explicit through the test what this value is actually always referring to, the current value of the revision field on the server.

Reply:
i renamed `rev` to `currentRevision`. i think passing in `expectedRevision` (instead of `currentRevision`+1) into kvstore methods is more clear so i kept that